golang/pkg/common/common.go

```diff
@@ -26,11 +31,40 @@ func GetAuthorizationKeysFromEnv() (auth.KeyPair, error) {
 	}
 
 	return auth.KeyPair{
-		ApiKey: apiKey,
+		ApiKey:    apiKey,
 		SecretKey: secretKey,
 	}, nil
 }
 
+func makeConnection(opts *args.CommonOptions, creds *auth.JwtPerRPCCredentials) (*grpc.ClientConn, error) {
+	if strings.HasSuffix(*opts.Endpoint, "443") {
+		var rootCAs *x509.CertPool
+		if *opts.CAfile != "" {
+			pemServerCA, err := os.ReadFile(*opts.CAfile)
+			if err != nil {
+				return nil, err
+			}
+
+			rootCAs = x509.NewCertPool()
+			if !rootCAs.AppendCertsFromPEM(pemServerCA) {
+				return nil, fmt.Errorf("failed to add server CA's certificate")
+			}
+		}
+
+		return grpc.Dial(
+			*opts.Endpoint,
+			grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{
+				RootCAs: rootCAs,
+			})),
+			grpc.WithPerRPCCredentials(creds),
+		)
+	}
+
+	return grpc.Dial(
+		*opts.Endpoint,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
```

Comment:
I have done the JWT credentials sending in the way that they are being sent only if the channel is secure (notice this part of code under if secure branch) https://github.com/Tinkoff/voicekit-examples/blob/ee21446134a411823d640a8c26ae3d4b039d19a4/golang/pkg/common/common.go#L68, so there is no reason to specify false there because JWT credentials are never sent through an insecure channel. What do you think, is it a good idea to also send credentials for the insecure channel?

Reply:
For some scenarious, it can be.